src/quo2/components/dropdowns/dropdown.cljs

```diff
@@ -1,165 +1,14 @@
-         :color  :main} "Dropdown"]
-       [icons/icon
-        (if @open?
-          (if dark?
-            :main-icons/pullup-dark
-            :main-icons/pullup)
-          (if dark?
-            :main-icons/dropdown-dark
-            :main-icons/dropdown))
-        {:size            20
-         :no-color        true
-         :container-style {:width         (+ icon-size 3)
-                           :border-radius 20
-                           :margin-left   (if (= :small size)
-                                            2
-                                            4)
-                           :margin-top    1
-                           :height        (+ icon-size 4)}}]])))
-
-(defn items-comp
-  [{:keys [items on-select]}]
-  (let [items-count (count items)]
-    [rn/scroll-view
-     {:horizontal          false
-      :nestedScrollEnabled true}
-     (doall
-      (map-indexed (fn [index item]
-                     [rn/touchable-opacity
-                      {:key      (str item index)
-                       :style    {:padding             4
-                                  :border-bottom-width (if (= index (- items-count 1))
-                                                         0
-                                                         1)
-                                  :border-color        (colors/theme-colors
-                                                        colors/neutral-100
-                                                        colors/white)
-                                  :text-align          :center}
-                       :on-press #(on-select item)}
-                      [text/text {:style {:text-align :center}} item]])
-                   items))]))
-
-(defn- f-dropdown
-  [{:keys [items icon text default-item on-select size disabled? border-color use-border? dd-color]}]
-  (let [dd-height (reanimated/use-shared-value 0)]
-    [rn/view {:style {:flex-grow 1}}
-     [dropdown-comp
-      {:items        items
-       :icon         icon
-       :disabled?    disabled?
-       :size         size
-       :dd-color     dd-color
-       :text         text
-       :border-color (colors/custom-color-by-theme border-color 50 60)
-       :use-border?  use-border?
-       :default-item default-item
-       :dd-height    dd-height}]
-     [reanimated/view
-      {:style (reanimated/apply-animations-to-style
-               {:height dd-height}
-               {})}
-      [items-comp
-       {:items     items
-        :on-select on-select}]]]))
+  (:require [quo2.components.buttons.button :as button]))
 
 (defn dropdown
-  [params]
-  [:f> f-dropdown params])
+  [_ _]
+  (fn [{:keys [on-change selected] :as opts} children]
+    [button/button
+     (merge
+      opts
+      {:after                   (if selected :i/pullup :i/dropdown)
+       :icon-secondary-no-color true
+       :pressed                 selected
+       :on-press                #((or on-change identity) selected)})
```

Comment:
We could make this code a bit clearer with a default destructure for `:on-change` or use `when`.

```clojure
#(when on-change (on-change selected))
```

Or

```clojure
(defn dropdown
  [_ _]
  (fn [{:keys [on-change selected] :or {on-change identity} :as opts} children]
    [button/button
     (merge
      opts
      {:after                   (if selected :i/pullup :i/dropdown)
       :icon-secondary-no-color true
       :pressed                 selected
       :on-press                #(on-change selected)})
     children]))
```

Reply:
yes thanks